starlark/library.go

```diff
@@ -588,12 +588,20 @@ func hasattr(thread *Thread, _ *Builtin, args Tuple, kwargs []Tuple) (Value, err
 
 // https://github.com/google/starlark-go/blob/master/doc/spec.md#hash
 func hash(thread *Thread, _ *Builtin, args Tuple, kwargs []Tuple) (Value, error) {
-	var x Value
-	if err := UnpackPositionalArgs("hash", args, kwargs, 1, &x); err != nil {
+	var s string
+	if err := UnpackPositionalArgs("hash", args, kwargs, 1, &s); err != nil {
 		return nil, err
 	}
-	h, err := x.Hash()
-	return MakeUint(uint(h)), err
+
+	// The Starlark spec requires that the hash function be
+	// deterministic across all runs (motivated by the need
+	// for reproducibility of builds). Thus we cannot call
+	// String.Hash, which may use the fastest implementation
+        // available, even if it varies across process restarts,
```

Comment:
Indentation is a little off here.

Reply:
Indentation fixed.

Implemented the algorithm specified in j.l.String docs.